### PR TITLE
Refactor/color picker

### LINKED ISSUE #1861: Make the `@yamada-ui/color-picker` component headless
script: make-component-headless
package: @yamada-ui/color-picker

### Description
Currently, the `@yamada-ui/color-picker` component has minimal styles set within the component's source code.
We want to make this style transferable to the theme and make the component headless, enabling more flexible UI customization.

### Problem Statement/Justification
By making the `@yamada-ui/color-picker` component headless, we can achieve more flexible styling tailored to the project.

### Proposed Solution or API
The styles within the component are mostly defined by `css: CSSUIObject` and set to a prop named `__css`. We will transfer these styles to the `packages/theme/src/components`.

The code below is just an example and not the actual code.

`packages/components/input/src/input-group.tsx`
```diff
const css: CSSUIObject = {
- width: "100%",
- display: "flex",
- position: "relative",
  ...styles.container,
}
```

`packages/theme/src/components/input.ts`
```diff
baseStyle: {
  container: {
+   width: "100%",
+   display: "flex",
+   position: "relative",
  },
  field: {
    width: "100%",
    minWidth: 0,
    outline: 0,
    position: "relative",
    appearance: "none",
    transitionProperty: "common",
    transitionDuration: "normal",
    _disabled: {
      opacity: 0.4,
      cursor: "not-allowed",
    },
  },
  element: {
    color: ["blackAlpha.600", "whiteAlpha.700"],
  },
},
```

> [!TIP]
>
> In cases where the styles within the component are dynamically changed, there may be instances where it is not possible to transfer the styles to the `theme`.
> In such cases, it needs to be discussed in this issue. Please feel free to send comments to this issue.

### Additional Information
The `@yamada-ui/color-picker` component exists in `packages/components/color-picker/src`. You need to check all the components in this folder and if they contain styles internally, you need to modify them.

This change could have a significant impact on projects using Yamada UI, so we will work on it within the scope of `v2.0`.
Please create the branch for working on this issue from the remote branch `v2.0`, and direct your pull requests to the `v2.0` branch as well.

Please select `major` for the changeset and create a pull request.

### PR DESCRIPTION
**Pull request by kurakke**
Closes #1861

## Description
make the color-picker component headless

## Is this a breaking change (Yes/No):
No

### CODE REVIEW

**Review by Hayato1031 — DISMISSED** [review]
LGTM

File: packages/components/color-picker/src/alpha-slider.tsx
Comment (Hayato1031) [inline packages/components/color-picker/src/alpha-slider.tsx:1]: Do you have any reason that you haven`t  change these code?
LINE 158~162
``
position: "absolute",
top: 0,
left: 0,
right: 0,
bottom: 0,
``

**Review by Hayato1031 — COMMENTED** [review]
Do you have any reason that you haven`t  change these code?
``packages/components/color-picker/src/alpha-slider.tsx``
LINE 158~162

``
position: "absolute",
top: 0,
left: 0,
right: 0,
bottom: 0,
``

**Comment by kurakke** [comment]
I fixed. Could you please review the changes?
@Hayato1031

File: packages/components/color-picker/src/saturation-slider.tsx
Comment (Hayato1031) [inline packages/components/color-picker/src/saturation-slider.tsx:1]: I missed these code.
Do you have any reason that you haven't change it?

LINE 76~95
```
const css: CSSUIObject = {
      position: "relative",
      _before: {
        content: `""`,
        display: "block",
        h: 0,
        pb: replaceObject(ratio, (r) => `${(1 / r) * 100}%`),
      },
      "& > *": {
        position: "absolute",
        top: "0",
        right: "0",
        bottom: "0",
        left: "0",
        display: "flex",
        justifyContent: "center",
        alignItems: "center",
        w: "100%",
        h: "100%",
      },
      ...styles.container,
      ...__css,
    }
```

**Comment by Hayato1031** [comment]
Test was failed.

Did you run `pnpm test` in your development environment?

**Comment by Hayato1031** [comment]
@kurakke 
Test was failed.

Did you run pnpm test in your development environment?

File: packages/components/color-picker/src/saturation-slider.tsx
Comment (kurakke) [reply packages/components/color-picker/src/saturation-slider.tsx:1]: Since @hirotomoyamada  told me that part is fine as it is, I have left it that way.

**Comment by kurakke** [comment]
> @kurakke Test was failed.
> 
> Did you run `pnpm test` in your development environment?

I apologize for forgetting to run the tests locally.
When running locally, the tests fail at different points than they do in the CI.
I am currently investigating the cause.

**Comment by kurakke** [comment]
@Hayato1031 
I have rebased the latest changes, and all tests have passed.
Could you please review and confirm?

**Review by Hayato1031 — APPROVED** [review]
LGTM

**Review by Neru-Neru — APPROVED** [review]

**Pull request merged**